### PR TITLE
[spinner] Add support for wayland backend

### PR DESCRIPTION
**Pull request by mariogrip**
This adds support for wayland backend to the spinner.

It currently defaults to mir if running on an android device and wayland
device if not.

Backend can be overritten by setting COMPOSITOR_BACKEND.

### CODE REVIEW

File: spinner/wayland/eglapp.cpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright © 2013-2018 Canonical Ltd.
+ *             2019 UBports project
```
Comment (UniversalSuperBox) [inline spinner/wayland/eglapp.cpp]: UBports Foundation, not project.

File: spinner/wayland/waylandegl.h [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    static void shell_surface_configure(void *data,
+        struct wl_shell_surface *wl_shell_surface,
+        uint32_t edges,
+        int32_t width,
+        int32_t height);
+    static void shell_surface_popup_done(void *data, struct wl_shell_surface *wl_shell_surface);
+
+};
+
+#endif //UNITYSYSTEMCOMPOSITOR_WAYLANDEGL_H
```
Comment (UniversalSuperBox) [inline spinner/wayland/waylandegl.h]: Several files are missing a trailing newline

File: spinner/eglspinner.cpp [hunk 3]
```diff
@@ -317,7 +335,41 @@ try
...
+    std::vector<std::shared_ptr<EglSurface>> surfaces;
+    char* backendEnv = getenv("COMPOSITOR_BACKEND");
+    Backend backend = UNKOWN;
+
+    if (backendEnv == nullptr) {
+        if (session_config.is_android()) {
+            printf("Is android device, defaults to mir\n");
+            backend = MIR;
+        } else {
+            printf("Is not android device, defaults to wayland\n");
```
Comment (UniversalSuperBox) [inline spinner/eglspinner.cpp]: Consider the following:

"Running on an Android device, using Mir output."

"Not running on an Android device, using Wayland output."

**Review by UniversalSuperBox — CHANGES_REQUESTED** [review]
I'm still not really sure between up or down on C++, but I know there are some stylistic, grammar, and licensing issues here.

**Comment by mariogrip** [comment]
Fixed :)

**Review by dobey — APPROVED** [review]
Looks ok and still compiles. Going to merge it since Dalton's comments were addressed and this is for edge.

**Pull request merged**